packages-ts/gauntlet-terra-contracts/src/lib/events.ts

```diff
@@ -0,0 +1,28 @@
+import { AccAddress } from '@terra-money/terra.js'
+
+type EventOraclePaid = {
+  transmitter: string
+  payee: string
+  amount: string
+  linkToken: string
+  method?: string
+}
+
+export const parseOraclePaidEvent = (event): EventOraclePaid => {
+  // Parse and validate every value
+  const transmitter = event.transmitter[0]
+  const payee = event.payee[0]
+  const amount = event.amount[0]
+  const link_token = event.link_token[0]
+
+  if (!AccAddress.validate(transmitter)) throw new Error(`Invalid transmitter address`)
```

Comment:
Throwing on invalid or undefined values seems a bit aggressive. Take a look at how we do it at the TS client: https://github.com/smartcontractkit/chainlink-terra/blob/develop/packages-ts/terra-sdk/src/index.ts#L50